src/python/pants/backend/python/tasks/gather_sources.py

```diff
@@ -68,10 +71,7 @@ def _get_pex_for_versioned_targets(self, interpreter, versioned_targets):
     if versioned_targets:
       target_set_id = VersionedTargetSet.from_versioned_targets(versioned_targets).cache_key.hash
     else:
-      # If there are no relevant targets, we still go through the motions of gathering
-      # an empty set of sources, to prevent downstream tasks from having to check
-      # for this special case.
-      target_set_id = 'no_targets'
+      raise TaskError('Tried to get pex for no targets in gather_sources')
```

Comment:
I think the message here could be better, in case an execution does end up here. Maybe

`Can't create pex in gather_sources: No relevant python targets provided`

Reply:
Done